File: src/my_project/easy_problems/from201to250/binary_tree_path.py
from typing import List, Optional 


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
        
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        
        if root is None: 
            return []

        def traversal(node, path):
            if node.left is None and node.right is None:
                answer.append("->".join(path))
                
            if node.left:
                path.append(str(node.left.val))
                traversal(node.left, path)
                path.pop()
                
            if node.right:
                path.append(str(node.right.val))
                traversal(node.right, path)
                path.pop()      
    
        answer = []
        path = [str(root.val)]
        traversal(root, path)
        return answer 
    
print("*".join(["1", "2"]))

 
 

File: my_project_typescript/easy_problems_typescript/from101to150/binaryTreePath.ts

//  * Definition for a binary tree node.
  class TreeNode {
      val: number
      left: TreeNode | null
      right: TreeNode | null
      constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
          this.val = (val===undefined ? 0 : val)
          this.left = (left===undefined ? null : left)
          this.right = (right===undefined ? null : right)
      }
  }


 function binaryTreePaths(root: TreeNode | null): string[] {

    if (root  === null){
        return []
    }

    function traversal(node: TreeNode, path: Array<string>){
        if (node.left === null && node.right === null) {
            answer.push(path.join("->"))
        }

        if (node.left){
            path.push(node.left.val.toString())
            traversal(node.left, path)
            path.pop()
        }
        if (node.right){
            path.push(node.right.val.toString())
            traversal(node.right, path)
            path.pop()
        }
    }


    let answer: Array<string> = []
    let path = [root.val.toString()]
    traversal(root, path)

    return answer

};





